src/plugins/vis_type_pie/public/to_ast.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getVisSchemas, VisToExpressionAst, SchemaConfig } from '../../visualizations/public';
+import { buildExpression, buildExpressionFunction } from '../../expressions/public';
+import { PieVisParams, LabelsParams } from './types';
+import { vislibPieName, VisTypePieExpressionFunctionDefinition } from './pie_fn';
+import { getEsaggsFn } from './to_ast_esaggs';
+
+const prepareDimension = (params: SchemaConfig) => {
+  const visdimension = buildExpressionFunction('visdimension', { accessor: params.accessor });
+
+  if (params.format) {
+    visdimension.addArgument('format', params.format.id);
+    visdimension.addArgument('formatParams', JSON.stringify(params.format.params));
+  }
+
+  return buildExpression([visdimension]);
+};
+
+const prepareLabels = (params: LabelsParams) => {
+  const pieLabels = buildExpressionFunction('pielabels', {
+    show: params.show,
+    last_level: params.last_level,
+    values: params.values,
+    truncate: params.truncate,
+  });
+  if (params.position) {
+    pieLabels.addArgument('position', params.position);
+  }
+  if (params.valuesFormat) {
+    pieLabels.addArgument('valuesFormat', params.valuesFormat);
+  }
+  if (params.percentDecimals) {
+    pieLabels.addArgument('percentDecimals', params.percentDecimals);
+  }
```

Comment:
This should be changed to something similar to:
```suggestion
  if (params.percentDecimals != null) {
    pieLabels.addArgument('percentDecimals', params.percentDecimals);
  }
```
To be able to keep the `0` as possible value from the slider

Reply:
Fixed https://github.com/elastic/kibana/pull/83929/commits/12abdd25cbbeddc7e0621624e8f668d2f7c4496d